client/me/account-close/closed.jsx

```diff
@@ -1,46 +1,104 @@
-import { Spinner } from '@automattic/components';
-import { localize } from 'i18n-calypso';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import EmptyContent from 'calypso/components/empty-content';
+import config from '@automattic/calypso-config';
+import { Button, Spinner } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import { connect, useDispatch } from 'react-redux';
+import { BlankCanvas } from 'calypso/components/blank-canvas';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { useSelector } from 'calypso/state';
+import { restoreAccount } from 'calypso/state/account/actions';
+import { getIsRestoring, getRestoreToken } from 'calypso/state/account/selectors';
 import getPreviousRoute from 'calypso/state/selectors/get-previous-route';
 import isAccountClosed from 'calypso/state/selectors/is-account-closed';
 
 import './closed.scss';
 
-class AccountSettingsClosedComponent extends Component {
-	onClick = () => {
-		window.location = '/';
-	};
+function AccountSettingsClosedComponent( { isUserAccountClosed } ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
 
-	render() {
-		const { isUserAccountClosed, translate } = this.props;
+	const isRestoring = useSelector( getIsRestoring );
 
-		if ( ! isUserAccountClosed ) {
-			return (
-				<div className="account-close__spinner">
-					<Spinner size={ 32 } />
-					<p className="account-close__spinner-text">
-						{ translate( 'Your account is being deleted' ) }
-					</p>
-				</div>
-			);
+	// restore token is either in the URL or in the reducer
+	const params = new URLSearchParams( window.location.search );
+	const urlToken = params.get( 'token' );
+	const storedToken = useSelector( getRestoreToken );
+	const restoreToken = urlToken || storedToken;
+
+	// Sync token to URL if not already there
+	useEffect( () => {
+		if ( storedToken && ! urlToken ) {
+			const newUrl = new URL( window.location.href );
+			newUrl.searchParams.set( 'token', storedToken );
+			window.history.replaceState( {}, '', newUrl.toString() );
 		}
+	}, [ storedToken, urlToken ] );
+
+	const onClick = () => {
```

Comment:
```suggestion
	const onReturnClick = () => {
```

Or maybe event `onCancelClick`?